runner/src/mill/runner/MillBuildRootModule.scala

```diff
@@ -61,20 +60,23 @@ class MillBuildRootModule()(implicit
 
   override def scalaVersion: T[String] = "2.13.10"
 
-  def scriptSources = T.sources {
-    MillBuildRootModule
-      .parseBuildFiles(millBuildRootModuleInfo)
-      .seenScripts
-      .keys
-      .map(PathRef(_))
-      .toSeq
+  /**
+   * All script files (that will get wrapped later)
+   * @see [[generateScriptSources]]
+   */
+  def scriptSources = T {
+    parseBuildFiles().seenScripts.keys.map(PathRef(_)).toSeq
   }
 
-  def parseBuildFiles = T {
-    scriptSources()
+  def parseBuildFiles: T[FileImportGraph] = T.input {
```

Comment:
Made it an input task, as it needs to reevaluate anyways, to detect changes. But this way, we can invert the dependency to `scriptSources` and avoid the second `parseBuildFiles(millBuildRootModuleInfo)` invocation.

Reply:
I'd like to avoid this being an input task so we don't need to re-parse everything every time you run something, or continuously over and over when you use `--watch`.

Running it twice redundantly in `scriptSources`/`parseBuildFiles` isn't great, but it's better than running it an unbounded number of times redundantly due to it being an input task

Reply:
Now, that you say it, I also recognize that we now run always whereas before we ran only if one of the returned sources changed. I'll revert.